lab_5_scrapper/scrapper.py

```diff
@@ -13,62 +52,92 @@ def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self.config_dto = self._extract_config_content()
+        self._validate_config_content()
+        self._get_seed_urls = self.get_seed_urls()
+        self._num_articles = self.get_num_articles()
+        self._get_headers = self.get_headers()
+        self._get_encoding = self.get_encoding()
+        self._get_timeout = self.get_timeout()
+        self._verify_certificate = self.get_verify_certificate()
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as infile:
+            reader = json.load(infile)
+        return ConfigDTO(**reader)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        if not self.config_dto.seed_urls or not isinstance(self.config_dto.seed_urls, list) \
+                or not all(isinstance(url, str) for url in self.config_dto.seed_urls):
+            raise IncorrectSeedURLError
+
+        if not isinstance(self.config_dto.total_articles_to_find_and_parse, int) \
+                or not 1 < self.config_dto.total_articles_to_find_and_parse < 150:
+            raise IncorrectNumberOfArticlesError
+
+        if not self.config_dto.seed_urls and not isinstance(self.config_dto.headers, dict) \
+                and not all(isinstance(key, str) and isinstance(value, str)
+                            for key, value in self.config_dto.headers.items()):
+            raise IncorrectHeadersError
+
+        if not isinstance(self.config_dto.encoding, str):
+            raise IncorrectEncodingError
+
+        if not isinstance(self.config_dto.timeout, int) or self.config_dto.timeout < 1 or self.config_dto.timeout > 60:
+            raise IncorrectTimeoutError
+
+        if not isinstance(self.config_dto.verify_certificate, bool):
+            raise IncorrectVerifyError
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self.config_dto.seed_urls
```

Comment:
If you get all information inside you getting methods using `self.config_dto.<something>`, so why do you define class attributes e.g. self._num_articles? How do you use  and get them, especially when these fields are protected (you use underscore in name)?

Try to reimplement this logic in another way - your getting methods e.g. `get_num_articles` should return `self._num_article` field value, as it is protected and there is no other way to get it.

Reply:
fbvddbd